commands/deployment_strategy.go

```diff
@@ -0,0 +1,75 @@
+package commands
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/cloudfoundry-incubator/multiapps-cli-plugin/util"
+	"strconv"
+)
+
+type DeploymentStrategy interface {
+	CreateProcessBuilder() *util.ProcessBuilder
+}
+
+type DeployCommandStrategy struct {
+}
+
+func (d *DeployCommandStrategy) CreateProcessBuilder() *util.ProcessBuilder {
+	processBuilder := util.NewProcessBuilder()
+	processBuilder.ProcessType(deployCommandProcessTypeProvider{}.GetProcessType())
+	return processBuilder
+}
+
+type BlueGreenCommandStrategy struct {
+	noConfirm  bool
+}
+
+func (b *BlueGreenCommandStrategy) CreateProcessBuilder() *util.ProcessBuilder {
+	processBuilder := util.NewProcessBuilder()
+	processBuilder.ProcessType(blueGreenDeployCommandProcessTypeProvider{}.GetProcessType())
+	processBuilder.Parameter("noConfirm", strconv.FormatBool(b.noConfirm))
+	processBuilder.Parameter("keepOriginalAppNamesAfterDeploy", strconv.FormatBool(true))
+	return processBuilder
+}
+
+func NewDeploymentStrategy(options map[string]interface{}, flags *flag.FlagSet, typeProvider ProcessTypeProvider) (DeploymentStrategy, error) {
+	if _, ok := typeProvider.(*blueGreenDeployCommandProcessTypeProvider); ok {
+		return &BlueGreenCommandStrategy{GetBoolOpt(noConfirmOpt, options)}, nil
+	}
+	if err := validateStrategyFlag(flags); err != nil {
+		return nil, err
+	}
+	strategy := GetStringOpt(strategyOpt, options)
+	if strategy == "default" || strategy == "" {
+		return &DeployCommandStrategy{}, nil
+	}
+	return &BlueGreenCommandStrategy{GetBoolOpt(noConfirmOpt, options)}, nil
+}
+
+func validateStrategyFlag(flags *flag.FlagSet) error {
+	var err error
+	flags.Visit(func(f *flag.Flag) {
+		if f.Name == strategyOpt {
+			if f.Value.String() == ""  {
+				err = errors.New("strategy flag defined but no argument specified")
+			} else if !containedIn(f.Value.String(), AvailableStrategies()) {
+				err = fmt.Errorf("%s is not a valid deployment strategy, available strategies: %v", f.Value.String(), AvailableStrategies())
+			}
+		}
+	})
+	return err
+}
+
+func AvailableStrategies() []string {
+	return []string{"blue-green", "default"}
+}
+
+func containedIn(s string, arr []string) bool {
```

Comment:
I think we already have such method -could you check in the other files?

Reply:
Yes, we have the same method in the util package.
Have renamed that one to be exported and deleted this one